application/src/main/java/org/togetherjava/tjbot/commands/help/OnGuildBanDeleteSpamThreadListener.java

```diff
@@ -0,0 +1,49 @@
+package org.togetherjava.tjbot.commands.help;
+
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.events.guild.GuildBanEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import org.jooq.Result;
+
+import org.togetherjava.tjbot.commands.EventReceiver;
+import org.togetherjava.tjbot.db.Database;
+import org.togetherjava.tjbot.db.generated.tables.records.HelpThreadsRecord;
+
+import java.time.Duration;
+import java.time.Instant;
+
+import static org.togetherjava.tjbot.db.generated.Tables.HELP_THREADS;
+
+/**
+ * Implements the functionality to delete recent help threads when the user gets banned.
+ */
+public final class OnGuildBanDeleteSpamThreadListener extends ListenerAdapter
+        implements EventReceiver {
+    private static final Duration recentThreadDuration = Duration.ofMinutes(30);
+
+    private final Database database;
+
+    /**
+     * @param database the database to get help threads from
+     */
+    public OnGuildBanDeleteSpamThreadListener(Database database) {
+        this.database = database;
+    }
+
+    @Override
+    public void onGuildBan(GuildBanEvent event) {
+        getRecentHelpThreads(event.getUser()).forEach(helpThreadsRecord -> event.getJDA()
+            .getThreadChannelById(helpThreadsRecord.getChannelId())
+            .delete()
+            .queue());
+    }
+
+    private Result<HelpThreadsRecord> getRecentHelpThreads(User user) {
+        return database
+            .read(context -> context.selectFrom(HELP_THREADS)
+                .where(HELP_THREADS.AUTHOR_ID.eq(user.getIdLong())
+                    .and(HELP_THREADS.CREATED_AT
+                        .greaterOrEqual(Instant.now().minus(recentThreadDuration)))))
```

Comment:
to improve readability, id move the math part of that instant stuff out into a variable.

```java
Instant recentThreadThreshold = Instant.now().minus(...);

return ...
    CREATED_AT.greater(recentThreadTreshold)
  ...
```
id also just use `greater`, its shorter to read and its not like we care whether > or >=

Reply:
fair. id still use it, but thats just NIT.

the other point is more important for me

Reply:
done